lang/resources/org/partiql/type-domains/partiql.ion

```diff
@@ -130,6 +130,9 @@ may then be further optimized by selecting better implementations of each operat
             (union setq::set_quantifier operands::(* expr 2))
             (except setq::set_quantifier operands::(* expr 2))
             (intersect setq::set_quantifier operands::(* expr 2))
+            (outer_union setq::set_quantifier operands::(* expr 2))
+            (outer_except setq::set_quantifier operands::(* expr 2))
+            (outer_intersect setq::set_quantifier operands::(* expr 2))
```

Comment:
Could there be a modeling of set operators in `partiql.ion` where the `OUTER` is included in the existing `union`, `except`, and `intersect` nodes rather than having to create other nodes? Not sure if that modeling would be cleaner and/or easier to work with than the current modeling of set operators.

Reply:
These are actually notably different operators. I would much prefer to make an abstraction here rather than how it is done today. These should all just be `(bagop op::bagop_type setq::set_quantifier operands::(* expr 2))`. It can get frustrating how little abstraction we have .. lots of duplication which continues to pile up

Reply:
As pointed out by Alan, there are existing issues to improve the IR
- https://github.com/partiql/partiql-lang-kotlin/issues/241 — NAry ops
- https://github.com/partiql/partiql-lang-kotlin/issues/239 — collection types